### ISSUE TITLE
Problem with default content group resolution when using ACL's

### ISSUE THREAD

**Issue by Dirklectisch** [zotonic/zotonic#1024]
I have a encountered a problem with determining the default content group for a context that is authorized using z_acl:sudo/1. This is especially a problem when used in combination with mod_import_anymeta since content might end up in the wrong group if ACL's have been configured beforehand.

The easiest way to reproduce is as follows.

1. Set a rule that allows 'users' all rights on a specific content group. (in my example rsc id 11064)

2. Determine the default_content_group.

    ```
    zotonic001@ginger)77> C.
    {context,,...}
    (zotonic001@ginger)81> m_rsc:rid(default_content_group, C).      
    325
    ```

3. See what the ACL thinks of as the default content group for an unauthorized user.

    ````
    (zotonic001@ginger)83> acl_user_groups_checks:default_content_group(m_rsc:rid(person, C), C).
    325
    ````

4. See what the ACL thinks of as the default content group for the super user.

    ````
    (zotonic001@ginger)84> acl_user_groups_checks:default_content_group(m_rsc:rid(person, z_acl:sudo(C)), z_acl:sudo(C)).
    11064
    ````

I would expect the default group to be 325 but somehow it is determined that the super user doesn't have any rights to put stuff in that group.

**Comment by ddeboer** [comment]
:+1:

**Comment by mworrell** [comment]
This might actually be correct behavior.
Which ACL rules did you define?

The system is checking against the ACL rules of the admin user, as:

```
z_acl:user(z_acl:sudo(z:c(mysite))).
1
```

**Comment by mworrell** [comment]
After discussion we can actually conclude that this is the correct behavior.
What is unexpected though is that the default group of the super user is the same as for a normal member. This needs to change.

**Comment by mworrell** [comment]
I think #1042 will make this behavior less unexpected.

Closing for now.

**Comment by mworrell** [comment]
Issue #1042 has been fixed in dab510c
